### ISSUE TITLE
Build failure: python310Packages.django-haystack

### ISSUE THREAD

**Issue by peti** [NixOS/nixpkgs#227395]
While attempting to update the machine that's running my Mailman server, the package `python310Packages.django-haystack` failed to build. The build seems to work fine at first -- it even runs the test suite successfully --, but then it hangs forever trying to shut down the test environment:

~~~
[...]
haystack/signals.py                                    34     34     0%
haystack/templatetags/__init__.py                       0      0   100%
haystack/templatetags/highlight.py                     63      3    95%
haystack/templatetags/more_like_this.py                58     48    17%
haystack/utils/__init__.py                             34     14    59%
haystack/utils/app_loading.py                          22     12    45%
haystack/utils/geo.py                                  30     23    23%
haystack/utils/highlighting.py                         93     11    88%
haystack/utils/loading.py                             201     58    71%
haystack/utils/log.py                                  14      7    50%
haystack/views.py                                     113      4    96%
-----------------------------------------------------------------------
TOTAL                                                4090   1531    63%
----------------------------------------------------------------------
Ran 432 tests in 8.653s

OK (SKIP=15)
[INFO/MainProcess] process shutting down
~~~

At this point, the build never finishes. I've reproduced this issues on three different machines with the current `nixos-unstable`.

**Comment by peti** [comment]
According to Hydra, the last successful build was in Nixpkgs version cc443761cbe4abfb6b0808c48de99622ffab0f14. I've tried to use "git bisect" to find the first failure of the build but didn't have much success because there is a stretch where many other packages are broken that are dependencies of python and these build failures might be hiding the underlying reason for the broken haystack package.

**Comment by peti** [comment]
This error blocks the NixOS Mailman module. Ping @qyliss & @ma27.

**Comment by kirillrdy** [comment]
bicected this to f228b9368b4c65775eb25dbb866fd5bba2ad3800

https://docs.python.org/release/3.10.10/whatsnew/changelog.html

has several mentions to `multiprocessing` module which is probably related to tests hanging at the end

**Comment by kirillrdy** [comment]
python 3.10.11 fixes the issues

https://github.com/NixOS/nixpkgs/pull/224913

you can test this on staging branch

**Comment by peti** [comment]
Appears to be fixed in 937a9d1ee7b1351d8c55fff6611a8edf6e7c1c37.
